### DISCUSSION TITLE
Reusable strings/Duplicate strings

### DISCUSSION THREAD

**Discussion by nandita1** [kwameopareasiedu/react-dialect#3] [Ideas]
@kwameopareasiedu I really liked this initiative!
But I have a question. What if we have duplicate strings in our codebase that we don't want to write again and again?
I can't make a constant with your library because it'll be treated as a variable right?

```
const DUPLICATE_STRING = `My name is {name}`
<Translate>{DUPLICATE_STRING}</Translate>

<Translate>{DUPLICATE_STRING}</Translate>
```

**Comment by kwameopareasiedu** [comment]
@nandita1 The way the CLI works is to use the content of the `<Translate>` tag as the key. Since this is just a single variable as the entire content, the key would be just a single variable and hence it could match any value. So I guess the answer is having a single variable as the content would be too little information for the translation to pick a specific value from the translations.

Thanks so much for bringing this edge case to my notice. I'm currently working on the new version to address the feedback I've gotten so far...
